torch/_functorch/_aot_autograd/runtime_wrappers.py

```diff
@@ -1771,9 +1777,15 @@ def backward(ctx, *flat_args):
                     len(CompiledFunction.metadata.output_types)
                     == num_flat_bw_args_with_grads
                 )
-                grad_output_types = [
-                    type(x) for x in all_args[-num_flat_bw_args_with_grads:]
-                ]
+
+                _l = -num_flat_bw_args_with_grads
+                _r = None
+                roffset = len(rng_args) + len(bw_tokens)
+                if roffset > 0:
+                    _l -= roffset
+                    _r = -roffset
+
+                grad_output_types = [type(x) for x in all_args[_l:_r]]
```

Comment:
ok what if.... instead of all of this, we just set `grad_output_types = [type(x) for x in flat_bw_args_with_grads]` (right before we `del flat_bw_args_with_grads` and lose access to it)?

All of this indexing hurts my brain :p